### PR TITLE
Profile editing

### PR DESCRIPTION
**Pull request by ridget**
* Profiles can now be edited by users
* profiles are accessible if the user opts-in
* style the sign in and sign out pages
* users can now login to edit their details

thanks to @xtrasimplicity for all his initial work.

screen shot 2017-03-17 at 10 29 39 pm

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,24 +1,54 @@
 class UsersController < Clearance::UsersController
+  def show
+    @user = User
+            .joins(:profile)
```
Comment (cattekin) [inline app/controllers/users_controller.rb]: I prefer 2 space indent for alignment, less noise when changing in the future.
i.e. if you changed `@user` to `@member`, you'd need to re-indent

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -21,7 +20,7 @@
 # directory. Alternatively, in the individual `*_spec.rb` files, manually
 # require only the support files necessary.
 #
-# Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }
+Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }
```
Comment (cattekin) [inline spec/rails_helper.rb]: Unsure on this. there is something nice about explicit requires, but does add overhead

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+require "rails_helper"
+
+RSpec.describe UsersController, type: :controller do
```
Comment (cattekin) [inline spec/controllers/users_controller_spec.rb]: make this a feature spec? :trollface:

File: app/views/users/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  </header>
+
+  <%= form_for @editing_member, url: user_path(@editing_member.user), as: :member, method: :patch do |form| %>
+    <%= render partial: '/users/form', object: form %>
+
+    <div class="submit-field">
+      <%= form.submit t('helpers.submit.editing_member.update'), class: 'btn' %>
+    </div>
+
+
```
Comment (cattekin) [inline app/views/users/edit.html.erb]: extra line

File: spec/models/profile_spec.rb [hunk 5]
```diff
@@ -10,4 +10,16 @@
     profile = Profile.new(user_id: nil)
     expect(profile).to_not be_valid
   end
+
+  describe ".visible_for_user" do
+    it "returns the matching users profile or visible profiles" do
+      current_user = create(:user)
```
Comment (cattekin) [inline spec/models/profile_spec.rb]: I have a preference for the explicit `FactoryGirl.create` syntax.
more explicit, less magic

Comment (ridget) [inline spec/models/profile_spec.rb]: Happy to stick to that.

File: spec/controllers/users_controller_spec.rb (hunk 3 above)
Comment (ridget) [inline spec/controllers/users_controller_spec.rb]: Torn on this, part of me feels we can just delete the whole thing as the scope just tests AR default behaviour and scope is tested elsewhere. I like having this sort of thing more as documentation around a requirement. That requirement word makes me agree on feature, but it's mainly the exception I care about. 

TLDR: feature it is :)

File: spec/rails_helper.rb (hunk 2 above)
Comment (ridget) [inline spec/rails_helper.rb]: was thinking more about newer contributors. There's overhead in doing this too if it gets big enough. I'll revert change for now since our spec size is pretty small
